waagent

```diff
@@ -550,7 +550,7 @@ class AbstractDistro(object):
         if os.path.isfile(mountpoint + "/swapfile") and os.path.getsize(mountpoint + "/swapfile") != (sizeKB * 1024):
             os.remove(mountpoint + "/swapfile")
         if not os.path.isfile(mountpoint + "/swapfile"):
-            Run("dd if=/dev/zero of=" + mountpoint + "/swapfile bs=1024 count=" + str(sizeKB))
+            self.fallocate(mountpoint + "/swapfile", sizeKB * 1024)
```

Comment:
Now I feel this one without any method things is better:
```Python
# fn = mountpoint + "/swapfile"
# We can also try using some fail-and-fallback methods here, like catching the 127 (command not found) return from fallocate and…
if sys.version_info >= (3,3):
  os.posix_fallocate(fn, 0, sizeKB * 1024) 
else if IsLinux():
  Run('fallocate -l ' + str(bytes) + ' ' + filename)
else:
  Run("dd if=/dev/zero of=" + fn + " bs=" + str(16384 * 1024) + " count=" + str(sizeKB / 16384))
  if sizeKB % 16384:
    Run("dd if=/dev/zero of=" + fn + " bs=" + str(sizeKB % 16384 * 1024) + " count=1")
```
Well, at least we don't need as much checks as for a real method~~, and all kinds of mess should be as short as possible so people are not that likely to see them~~.